### DISCUSSION TITLE
How to change the sort of RGBA chanels before BLImage reading image using readFromFile.

### DISCUSSION THREAD

**Discussion by jiangnanfengyu** [blend2d/blend2d#215] [Q&A] [answered]
Hello, I have a problem. I tried to show a image by using another image lib, raylib, after I load a image by BLImage.readFromFile and pass the pixeData to raylib. However, the raylib has a different sort of RGBA, I want to know how to change the sort of RGBA chanels before BLImage reading the image. I find the BLFormatInfo.setShifts, but I didn't find the method to use it in BLImage class.

**Comment by Sosukodo** [comment]
That's a good question..  

How might one go about converting BL_FORMAT_PRGB32  to some RGBA format?

**Reply by jiangnanfengyu** [reply]
Thanks.

**Comment by kobalicek** [comment, marked as answer]
If the format you are using is not compatible with built-in Blend2D formats, you have to either us BLPixelConverter to convert the data or to use your own function that converts the pixels.

**Reply by jiangnanfengyu** [reply]
Thanks.
